libvirt : host cpu model and features detection (#357)
canonical/ubuntu-server-documentation: how-to/virtualisation/libvirt.md

Some useful documentation on the constraints and considerations of live migration can be found at the [Ubuntu Wiki](https://wiki.ubuntu.com/QemuKVMMigration).

## CPU model and topology

libvirt abstracts CPU configuration and provides several options to specify the VM CPU model in the domain definition:

 - custom
 - host-model
 - host-passthrough
 - maximum

> While most of thee modes are straightforward, the behavior of `host-model` is more subtle. There is no direct

Reviewer: ```suggestion
While most of thee modes are straightforward, the behavior of `host-model` is more subtle and the source of many common misunderstandings. There is no direct
```

translation of `host-model` into a single QEMU `-cpu` argument. Instead, libvirt selects a baseline CPU model and appends
a list of features:

> ```bash

Reviewer: ```suggestion
```bash
# example snippet from qemu command line generated by libvirt
```

> -cpu Haswell-noTSX-IBRS,vmx=on,pdcm=off,...,vmx-entry-load-efer=on,vmx-eptp-switching=on

Reviewer: ```suggestion
... -cpu Haswell-noTSX-IBRS,vmx=on,pdcm=off,...,vmx-entry-load-efer=on,vmx-eptp-switching=on ...
```

```

> Because libvirt may not include every known CPU model, it chooses the model that shares the largest set of features with

Reviewer: ```suggestion
Because libvirt can not include every known CPU model and variant, it chooses the model that shares the largest set of features with
```

> the host’s physical CPU and then lists the remaining features explicitly. In many cases, libvirt does not detect the

Reviewer: ```suggestion
the host’s physical CPU and then lists the remaining features explicitly. In many cases, libvirt therefore can not detect the
```
